rsocket/internal/ClientResumeStatusCallback.h

```diff
@@ -6,22 +6,29 @@
 
 namespace rsocket {
 
+// Thrown when an ERROR frame with CONNECTION_ERROR is received during
+// resuming operation.
+// TODO: in this case we should get the DuplexConnection back to
+// create a new instance of RS with it
+class ResumptionException : public std::runtime_error {
```

Comment:
Agreed.  I will make those changes in a separate PR.  The focus is now mainly on the user-facing API.